infrastructure/constructs/bundled_code.py

```diff
@@ -1,14 +1,73 @@
-from os.path import join
+from pathlib import Path
+from re import sub
+from subprocess import PIPE, check_call, run
+from sys import executable
 
 from aws_cdk import BundlingOptions, aws_lambda
 
 from .backend import BACKEND_DIRECTORY
 from .lambda_config import PYTHON_RUNTIME
 
 
-def bundled_code(directory: str) -> aws_lambda.Code:
+def poetry_export_extras(packaging_directory: str, lambda_directory: str) -> None:
+    # There isn't an elegant way of getting poetry to install package dependencies in a bespoke
+    # target lambda_directory within Python, so we export a requirements file and install using pip.
+    # This has been raised and discussed by the community as below:
+    # https://github.com/python-poetry/poetry/issues/1937
+
+    export_extras = run(
+        ["poetry", "export", f"--extras={lambda_directory}", "--without-hashes"],
+        check=True,
+        stdout=PIPE,
+    ).stdout
+    # Remove botocore as this is already installed in the lambda layer
+    export_extras = sub(b"botocore==.*\n", b"", export_extras)
+
+    with open(
+        f"{packaging_directory}/{lambda_directory}/requirements.txt", "wb"
+    ) as lambda_requirements:
+        lambda_requirements.write(export_extras)
+
+
+def pip_install_requirements(packaging_directory: str, lambda_directory: str) -> None:
+    # Documentation recommend against calling pip internal api; rather, via command line
+    # https://pip.pypa.io/en/latest/user_guide/#using-pip-from-your-program
+
+    # Target already exists warning is discussed here
+    # https://github.com/pypa/pip/issues/8063
+    check_call(
+        [
+            executable,
+            "-m",
+            "pip",
+            "install",
+            "--no-deps",
+            "--quiet",
+            f"--cache-dir={packaging_directory}/cache",
+            f"--requirement={packaging_directory}/{lambda_directory}/requirements.txt",
+            f"--target={packaging_directory}/{lambda_directory}",
+        ]
+    )
+
+
+def bundled_code(lambda_directory: str) -> aws_lambda.Code:
+    packaging_directory = (
+        f"{BACKEND_DIRECTORY}/.lambda_out"  # Temp working directory storing lambda packages
+    )
+    Path(f"{packaging_directory}/{lambda_directory}").mkdir(parents=True, exist_ok=True)
+    poetry_export_extras(packaging_directory, lambda_directory)
+    pip_install_requirements(packaging_directory, lambda_directory)
     bundling_options = BundlingOptions(
         image=PYTHON_RUNTIME.bundling_image,  # pylint:disable=no-member
-        command=[join(BACKEND_DIRECTORY, "bundle.bash"), directory],
+        command=[
+            "bash",
+            "-c",
+            "mkdir --parents /asset-output/geostore/$1 && \
+                  cp --archive --update $0/$1/* /asset-output/ && \
+                  cp --archive --update /asset-input/geostore/*.py /asset-output/geostore/ && \
+                  cp --archive --update /asset-input/geostore/$1 /asset-output/geostore/ ",
```

Comment:
This line can be merged with the previous, since they both have the same target.